src/unaudited/ECDSAStakeRegistry.sol

```diff
@@ -0,0 +1,385 @@
+    /// @param _dataHash The hash of the data that is signed.
+    /// @param _signature The signature to validate.
+    function _validateSignature(
+        address _signer,
+        bytes32 _dataHash,
+        bytes memory _signature
+    ) internal view {
+        if (!_signer.isValidSignatureNow(_dataHash, _signature)) revert InvalidSignature();
+    }
+
+    /// @notice Retrieves the operator weight for a signer, either at the last checkpoint or a specified block.
+    /// @param _signer The address of the signer whose weight is returned.
+    /// @param _referenceBlock The block number to query the operator's weight at, or the maximum uint32 value for the last checkpoint.
+    /// @return The weight of the operator.
+    function _getOperatorWeight(
+        address _signer,
+        uint32 _referenceBlock
+    ) internal view returns (uint256) {
+        if (_referenceBlock == type(uint32).max) {
+            return _operatorWeightHistory[_signer].latest();
+        } else {
+            return _operatorWeightHistory[_signer].getAtBlock(_referenceBlock);
+        }
+    }
+
+    /// @notice Retrieve the total stake weight at a specific block or the latest if not specified.
+    /// @dev If the `_referenceBlock` is the maximum value for uint32, the latest total weight is returned.
+    /// @param _referenceBlock The block number to retrieve the total stake weight from.
+    /// @return The total stake weight at the given block or the latest if the given block is the max uint32 value.
+    function _getTotalWeight(uint32 _referenceBlock) internal view returns (uint256) {
+        if (_referenceBlock == type(uint32).max) {
+            return _totalWeightHistory.latest();
+        } else {
+            return _totalWeightHistory.getAtBlock(_referenceBlock);
+        }
+    }
+
+    /// @notice Retrieves the threshold stake for a given reference block.
+    /// @param _referenceBlock The block number to query the threshold stake for.
+    /// If set to the maximum uint32 value, it retrieves the latest threshold stake.
+    /// @return The threshold stake in basis points for the reference block.
+    function _getThresholdStake(uint32 _referenceBlock) internal view returns (uint256) {
+        if (_referenceBlock == type(uint32).max) {
+            return _thresholdWeightBpsHistory.latest();
+        } else {
+            return _thresholdWeightBpsHistory.getAtBlock(_referenceBlock);
+        }
+    }
+
+    /// @notice Validates that the cumulative stake of signed messages meets or exceeds the required threshold.
+    /// @param _signedWeight The cumulative weight of the signers that have signed the message.
+    /// @param _referenceBlock The block number to verify the stake threshold for
+    function _validateThresholdStake(uint256 _signedWeight, uint32 _referenceBlock) internal view {
+        uint256 totalWeight = _getTotalWeight(_referenceBlock);
+        if (_getThresholdStake(_referenceBlock) > (_signedWeight * BPS) / totalWeight)
```

Comment:
probably want non-strict inequality here?
also given the usage of `_thresholdWeightBpsHistory` you might want to sanitize those inputs more, so the threshold can't exceed 100%

Reply:
I think the inequality choice is correct.

if `signedWeight` == `thresholdWeight`, then don't revert
if `signedWeight` > `thresholdWeight`, then don't revert
if `signedWeight` < `thresholdWeight`, then revert

For sanitization, do you mean make sure that `_signedWeight` is less than or equal to `totalWeight`?

Reply:
It seems like this should only be a concern in the "out of sync" scenario discussed for updating the multipliers on strategies within a quorum and not subsequently updating all the operators' weights.  I will add the check against totalWeight here, but it seems like there also needs to be an update to all the operators when the quorum changes to make sure there is never more `signedWeight` than `totalWeight`

Reply:
I again read this wrong, totally my bad.
for input sanitization I believe I was thinking that _updateStakeThreshold needs to make sure its input is always less than or equal to `BPS` -- my apologies for the lack of clarity there -- I think this is the right place to sanitize:
https://github.com/Layr-Labs/eigenlayer-middleware/blob/f723e4f61d4985e7c706564f023b32f51af30f0f/src/unaudited/ECDSAStakeRegistry.sol#L196-L199

Reply:
makes sense, yeah this should have a check to make sure the new `_thresholdWeightBps` is less than `BPS`

Reply:
improved the threshold weight usage in this commit https://github.com/Layr-Labs/eigenlayer-middleware/pull/194/commits/e360e91a54ed668fd5311b03edc68436b47884b6